packages/lesswrong/components/posts/PostsEditForm.tsx

```diff
@@ -58,11 +58,7 @@ const PostsEditForm = ({ documentId, classes }: {
 
   // If we only have read access to this post, but it's shared with us,
   // redirect to the collaborative editor.
-  if (document
-    && document.userId!==currentUser._id
-    && document.sharingSettings
-    && !userCanDo(currentUser, 'posts.edit.all')
-  ) {
+  if (document && !canUserEditPost(currentUser, document) && document.sharingSettings) {
```

Comment:
What is the check for `document.sharingSettings` achieving here?

Reply:
So this predates me and I'm not 100% sure of the goals, but I think the idea is that you should only end up on the PostCollaborationEditor if the post actually has sharing settings.